File: jmlparser-jml-tools/src/main/java/com/github/jmlparser/lint/rules/MethodBodyHasNoContract.java
package com.github.jmlparser.lint.rules;

import com.github.javaparser.ast.body.MethodDeclaration;
import com.github.javaparser.ast.validator.ProblemReporter;
import com.github.javaparser.ast.validator.VisitorValidator;

public class MethodBodyHasNoContract extends VisitorValidator {
    @Override
    public void visit(MethodDeclaration n, ProblemReporter arg) {
        if (n.getBody().isPresent() && n.getBody().get().getContracts().isPresent()
                && !n.getBody().get().getContracts().get().isEmpty()
        ) {
            arg.report(n, "Body of method has a block contract.");
        }
        super.visit(n, arg);
    }
}

File: tools/lint/src/main/kotlin/io/github/jmltoolkit/lint/rules/MethodBodyHasNoContract.kt
package io.github.jmltoolkit.lint.rules

import com.github.javaparser.ast.body.MethodDeclaration
import com.github.javaparser.ast.validator.ProblemReporter
import com.github.javaparser.ast.validator.VisitorValidator

class MethodBodyHasNoContract : VisitorValidator() {
    override fun visit(n: MethodDeclaration, arg: ProblemReporter) {
        if (n.body.isPresent && !n.body.get().contracts.isEmpty()) {
            arg.report(n, "Body of method has a block contract.")
        }
        super.visit(n, arg)
    }
}
